test/task-and-workflow-proxies.test.js

```diff
@@ -0,0 +1,13 @@
+const testProfiles = require('./helpers/testProfiles')
+
+testProfiles('Task and workflow proxies', async (t, {executeWorkflow}) => {
+  const execution = await executeWorkflow({
+    workflowId: 'task-and-workflow-proxies',
+    type: 'TaskAndWorkflowProxies',
+  })
+
+  const {taskResult, childworkflowResult} = await execution.promise
+
+  t.is(taskResult, 'task-success', 'yields the task result')
+  t.is(childworkflowResult, 'child-workflow-success', 'yields the task result')
+})
```

Comment:
Normally I find reading the tests gives me some clue as what going on, especially when an interface of a module is not clear. This test however gives me nothing, in fact I think I am more confused. 

There is no module called proxy, nothing in the documentation talking about proxies, so when you say your testing proxies, what are you testing?

The best guess I can come up with is that it has something to do with child workflows, since that is at least used in the task-and-workflow-proxies workflow.

Reply:
The task and workflow proxy functions in the DeciderContext.  The test could have been called DecisionContext workflow and task function proxies, be my guest to PR it :)

Reply:
If you open the workflow that is part of the test, you will see what kind of behaviour it is testing.

Reply:
I think your struggling to see this from the view of someone who has not written this code.

https://github.com/Skalar/soflow/search?utf8=%E2%9C%93&q=proxies&type=

Proxies is just a variable name in some internal code, it seems odd to me to name a test after it, and then expect other developers to come along and understand what functionality is being tested, and more importantly what the intended behaviour is supposed to be.

I already read the workflow used by the test and that is why I think it might have something to to with child workflows, unless there is some subtly different about that workflow I am not noticing?

Reply:
Task/Workflow proxies is just a (perhaps badly chosen) name to describe the higher level concept of being able to run tasks and child workflows within a workflow decider, by invoking taskName() and workflowName(), as opposed to using the task() and childWorkflow() actions.

The taskName() and workflowName() functions are mere proxies to these action.